### PR TITLE
Add byte size parameter for number of data bits of serial port communication

### LINKED ISSUE #368: Serial port change port settings (number of data bits)
We would like to use Modelica_DeviceDrivers to send data through a serial port.
The receiving device uses the serial port settings as shown in the image below, in particular 7 data bits.

serielport_settings

We use the MDD component `Modelica_DeviceDrivers.Blocks.Communication.SerialPortSend`.
Is there a way to change the number of data bits within the MDD Library?
So far we haven't found a way to change this parameter to 7 (instead of 8, which seems to be used by the MDD Library as standard value).

We tried to change the header file by setting `dcb.ByteSize = 7;`, but with no success.
https://github.com/modelica-3rdparty/Modelica_DeviceDrivers/blob/8267f35ef33f0893e65df5a3c047514936df6afd/Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h#L202-L212

How could we approach this problem?
Do we have to change the header file and recompile the files within the Modelica_DeviceDrivers 2.1.1\Resources folder?
We appreciate any help, as our knowledge with c-code compilation is limited.. : s

### PR DESCRIPTION
**Pull request by tbeu**
Closes #368.

@sertho @max-privato FYI

Reminder: ToDo before v2.2.0 release: Recompile ITI_MDD.dll.

### CODE REVIEW

File: Modelica_DeviceDrivers/Blocks/Communication.mo [hunk 1]
```diff
@@ -270,24 +271,24 @@ See <a href=\"modelica://Modelica_DeviceDrivers.Blocks.Examples.TestSerialPackag
...
-      "Serial port baud rate"
-       annotation (Dialog(group="Outgoing data"));
-    parameter Integer parity = 0
-      "set parity (0 - no parity, 1 - even, 2 - odd)"
-       annotation (Dialog(group="Outgoing data"));
+    parameter SerialBaudRate baud = SerialBaudRate.B9600 "Serial port baud rate"
+      annotation (Dialog(group="Outgoing data"));
+    parameter Integer parity = 0 "set parity (0 - no parity, 1 - even, 2 - odd)"
+      annotation (Dialog(group="Outgoing data"));
+    parameter Integer byteSize(min=5, max=8) = 8 "Byte size"
```
Comment (bernhard-thiele) [inline Modelica_DeviceDrivers/Blocks/Communication.mo:282]: Maybe add a bit more explanation. Although the term `byteSize` is what is used in serial port programming, it just feels strange to specify that a byte has a different number of bits than 8. Maybe following helps a "non-expert" user:
"Number of data bits transmitted per (serial data format) byte  (8 is most common)"

File: Modelica_DeviceDrivers/Communication/SerialPort.mo [hunk 2]
```diff
@@ -12,8 +12,9 @@ encapsulated function constructor
   input Integer parity = 0 "0 - no parity, 1 - even, 2 - odd";
   input Integer receiver = 1 "0 - sender, 1 - receiver";
   input SerialBaudRate baud;
+  input Integer byteSize = 8 "Byte size";
```
Comment (bernhard-thiele) [inline Modelica_DeviceDrivers/Communication/SerialPort.mo:15]: Like above I would prefer a more elaborate description string.

**Review by bernhard-thiele — CHANGES_REQUESTED** [review]
As commented in the code, I think there should be a bit more elaborate description strings. The term  "byte size" in this context is different than the usual meaning.

Since threre are only 4 valid values, one could also use an enumeration, but that is a matter  of taste and I'm also fine with just keeping it as a simple Integer.

**Comment by tbeu** [comment]
> As commented in the code, I think there should be a bit more elaborate description strings. The term "byte size" in this context is different than the usual meaning.

Resolved as proposed.

> Since threre are only 4 valid values, one could also use an enumeration, but that is a matter of taste and I'm also fine with just keeping it as a simple Integer.

Yes, there are 4 possible values in the Linux case. But I did not figure out if that also is true for the Win implementation. I used Integer type for simplicity here (similar as parity which holds three different alternatives only).

**Review by bernhard-thiele — APPROVED** [review]

**Pull request merged**